pkg/table/tables/tables.go

```diff
@@ -1163,13 +1163,22 @@ func (t *TableCommon) removeRecord(ctx table.MutateContext, txn kv.Transaction,
 	if m.IsActiveActive {
 		// For active-active replication, if _tidb_origin_ts is not null, it means the row is updated by upstream TiDBs.
 		// Then this txn must make sure its commit ts is greater than the _tidb_origin_ts. (last-write-win conflict resolving strategy, so-called LWW)
-		for _, col := range t.Columns {
-			if col.State == model.StatePublic && col.Name == model.ExtraOriginTSName {
-				value := r[col.Offset]
-				if !value.IsNull() {
-					txn.SetOption(kv.CommitWaitUntilTSO, value.GetUint64())
+		extraOriginTSOffset := opt.GetExtraOriginTSOffset()
+		if extraOriginTSOffset >= 0 {
+			value := r[extraOriginTSOffset]
+			if !value.IsNull() {
+				txn.SetOption(kv.CommitWaitUntilTSO, value.GetUint64())
+			}
+		} else {
+			// When no pruning happened, directly use Column.Offset to find _tidb_origin_ts column in the row.
+			for _, col := range t.Columns {
```

Comment:
If the columns are pruned but the code forget to set `extraOriginTSOffset`, could we recorgnize it and return an error?